src/stdlib_stringlist.f90

```diff
@@ -0,0 +1,1063 @@
+    type(string_type), intent(in)     :: string
+    type(stringlist_type)             :: append_string
+
+    append_string = list
+    call append_string%insert( list_tail, string )
+
+end function append_string
+
+pure function prepend_char( string, list )
+    character(len=*), intent(in)      :: string
+    type(stringlist_type), intent(in) :: list
+    type(stringlist_type)             :: prepend_char
+
+    prepend_char = string_type( string ) // list
+
+end function prepend_char
+
+pure function prepend_string( string, list )
+    type(string_type), intent(in)     :: string
+    type(stringlist_type), intent(in) :: list
+    type(stringlist_type)             :: prepend_string
+
+    prepend_string = list
+    call prepend_string%insert( list_head, string )
+
+end function prepend_string
+
+pure function append_stringlist( list, slist )
+    type(stringlist_type), intent(in) :: list
+    type(stringlist_type), intent(in) :: slist
+    type(stringlist_type)             :: append_stringlist
+
+    append_stringlist = list
+    call append_stringlist%insert( list_tail, slist )
+
+end function append_stringlist
+
+pure function append_stringarray( list, sarray )
+    type(stringlist_type), intent(in)          :: list
+    character(len=*), dimension(:), intent(in) :: sarray
+    type(stringlist_type)                      :: append_stringarray
+
+    append_stringarray = list
+    call append_stringarray%insert( list_tail, sarray )
+
+end function append_stringarray
+
+pure function prepend_stringarray( sarray, list )
+    character(len=*), dimension(:), intent(in) :: sarray
+    type(stringlist_type), intent(in)          :: list
+    type(stringlist_type)                      :: prepend_stringarray
+
+    prepend_stringarray = list
+    call prepend_stringarray%insert( list_head, sarray )
+
+end function prepend_stringarray
+
+
+! destroy_list --
+!     Destroy the contents of the list
+!
+! Arguments:
+!     list                   The list of strings in question
+!
+subroutine destroy_list( list )
+    type(stringlist_type), intent(out) :: list
+
+    list%size = 0
+    deallocate( list%stringarray )
+
+end subroutine destroy_list
+
+! length_list --
+!     Return the size (length) of the list
+!
+! Arguments:
+!     list                   The list of strings to retrieve the string from
+!
+pure integer function length_list( list )
+    type(stringlist_type), intent(in) :: list
+
+    length_list = list%size
+
+end function length_list
+
+pure integer function capacity( list )
+    type(stringlist_type), intent(in) :: list
+
+    capacity = 0
+    if ( allocated( list%stringarray ) ) then
+        capacity = size( list%stringarray )
+    end if
+
+end function capacity
+
+! Convert fidx and bidx to the equivalent integer
+pure integer function to_idxn( idx )
+    type(stringlist_index_type), intent(in) :: idx
+    integer                                 :: to_idxn
+
+    to_idxn = merge( idx%offset, list%size + 2 - idx%offset, idx%head )
```

Comment:
"Symbol 'list' has no IMPLICIT type"

Reply:
my bad!, if you wish you can suggest these changes using github to get the contribution to your name.